llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -75,10 +76,13 @@ RISCVTargetLowering::RISCVTargetLowering(const TargetMachine &TM,
     ABI = Subtarget.is64Bit() ? RISCVABI::ABI_LP64 : RISCVABI::ABI_ILP32;
   }
 
-  if (ABI == RISCVABI::ABI_CHERIOT)
+  if (ABI == RISCVABI::ABI_CHERIOT) {
     for (int LCID = 0; LCID < RTLIB::UNKNOWN_LIBCALL; ++LCID)
-      setLibcallCallingConv(static_cast<RTLIB::Libcall>(LCID),
-                            CallingConv::CHERI_LibCall);
+      setLibcallCallingConv(static_cast<RTLIB::Libcall>(LCID), CallingConv::CHERI_LibCall);
+  } else if (ABI == RISCVABI::ABI_CHERIOT_BAREMETAL) {
+    for (int LCID = 0; LCID < RTLIB::UNKNOWN_LIBCALL; ++LCID)
+      setLibcallCallingConv(static_cast<RTLIB::Libcall>(LCID), CallingConv::CHERI_CCall);
+  }
```

Comment:
Is this right?  I don't think we want libcalls to use the cross-compartment calling convention in bare-metal mode.

Reply:
+1 CallingConv::CC_C seems to be the correct one?